src/Operators/finitedifference.jl

```diff
@@ -1459,6 +1459,139 @@ end
     return Geometry.Contravariant3Vector(vᶠ * a)
 end
 
+"""
+    U = FCTBorisBook(;boundaries)
+    U.(v, x)
+
+Correct the flux using the flux-corrected transport formulation by Boris and Book
+Input arguments:
+- a face-valued vector field `v`
+- a center-valued field `x`
+```math
+U(v,x)[i] = \\begin{cases}
+  v[i] \\left(-2 x[i-\\tfrac{3}{2}] + 10 x[i-\\tfrac{1}{2}] + 4 x[i+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+  v[i] \\left(4 x[i-\\tfrac{1}{2}] + 10 x[i+\\tfrac{1}{2}] -2 x[i+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
+  \\end{cases}
+```
+This stencil is based on [BorisBook1973](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, with a [`SetValue`](@ref) boundary.
+"""
+struct FCTBorisBook{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTBorisBook(; kwargs...) = FCTBorisBook(NamedTuple(kwargs))
+
+return_eltype(::FCTBorisBook, V, A) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceFiniteDifferenceSpace,
+    arg_space::Spaces.CenterFiniteDifferenceSpace,
+) = velocity_space
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    arg_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = velocity_space
+
+@inline function fct_boris_book(v, a⁻, a⁺, a⁺⁺, a⁺⁺⁺, step)
+    if v != zero(eltype(v))
+        sign(v) ⊠ (RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                RecursiveApply.rmap(abs, v),
+                RecursiveApply.rmap(
+                    min,
+                    sign(v) ⊠ (a⁺⁺⁺ - a⁺⁺) ⊠ step,
+                    sign(v) ⊠ (a⁺ - a⁻) ⊠ step,
+                ),
+            ),
+        ))
+    else
+        RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                v,
+                RecursiveApply.rmap(min, (a⁺⁺⁺ - a⁺⁺) ⊠ step, (a⁺ - a⁻) ⊠ step),
+            ),
+        )
+    end
+end
+
+stencil_interior_width(::FCTBorisBook, velocity, arg) =
+    ((0, 0), (-half, half + 2))
+
+@inline function stencil_interior(::FCTBorisBook, loc, idx, hidx, velocity, arg)
+    space = axes(arg)
+    a⁻ = getidx(arg, loc, idx - half, hidx)
+    a⁺ = getidx(arg, loc, idx + half, hidx)
```

Comment:
Does `a⁻` represent `phi_{j-1}^{td}`, why's it idx - half instead of idx - half - 1 ? Is the idx  j+1/2 ?

Reply:
`a⁺` is `phi_{j}^{td}`, and `a⁻` is `phi_{j-1}^{td}`, similar to what we have for the 3rd-order-upwinding operator, see this [line](https://github.com/CliMA/ClimaCore.jl/blob/b133c3ae65e40c993f46c57466563d8cd1f99faa/src/Operators/finitedifference.jl#L1354)

Reply:
@Zhengyu-Huang, another way to think about this is that `PlusHalf(0)` comes before `1`.

Reply:
I still need some help to understand the code here, since it is different from 3rd order upwinding stencils, so I can implement Zalesak Corrector

1. Does this function compute the flux on the `idx` face, and the faces are labeled as 1, 2, ..., 
2. a is a cell center variable, phi^{td}, and they are labeled as 1, 2, ...
3. for face j, the cells on left and right are labeled as j-1 and j
4. to compute flux on the jth face, we need, phi^{td}[j-2], phi^{td}[j-1], phi^{td}[j], phi^{td}[j+1]
5. `idx - half` in the code means `idx - 2` here?

Reply:
See #885